resources/constraint_templates/warn_service_account_secret_delete.yaml

```diff
@@ -0,0 +1,25 @@
+apiVersion: templates.gatekeeper.sh/v1
+kind: ConstraintTemplate
+metadata:
+  name: k8swarnserviceaccountsecretdelete
+  annotations:
+    metadata.gatekeeper.sh/title: Warn Service Account Secret delete
+    description: This policy warns any time a service account secret token is deleted with the type "kubernetes.io/service-account-token"
+spec:
+  crd:
+    spec:
+      names:
+        kind: k8swarnserviceaccountsecretdelete
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package k8swarnserviceaccountsecretdelete
+
+        operations = {"DELETE"}
+
+        violation[{"msg": msg}] {
+          input.review.kind.kind == "Secret"
+          input.review.object.type == "kubernetes.io/service-account-token"
+          operations[input.review.operation]
+          msg := "Since Kubernetes 1.24, he ability to rotate service account tokens by deleting the secret has been deprecated. Cloud Platform recommends using the `cloud-platform-terraform-serviceaccount` module instead. You can find more info here - https://user-guide.cloud-platform.service.justice.gov.uk/documentation/other-topics/rotate-tokens-keys.html"
```

Comment:
```suggestion
          msg := "Since Kubernetes 1.24, the ability to rotate service account tokens by deleting the secret has been deprecated. Cloud Platform recommends using the `cloud-platform-terraform-serviceaccount` module instead. You can find more info here - https://user-guide.cloud-platform.service.justice.gov.uk/documentation/other-topics/rotate-tokens-keys.html"
```

Reply:
Added 👍